cloud_experts_tutorials/cloud-experts-deploying-application/cloud-experts-deploying-application-logging.adoc

```diff
@@ -0,0 +1,177 @@
+:_mod-docs-content-type: ASSEMBLY
+[id="cloud-experts-deploying-application-logging"]
+= Tutorial: Logging
+include::_attributes/common-attributes.adoc[]
+include::_attributes/attributes-openshift-dedicated.adoc[]
+:context: cloud-experts-deploying-application-logging
+
+toc::[]
+
+//rosaworkshop.io content metadata
+//Brought into ROSA product docs 2024-05-01
+
+There are various methods to view your logs in {product-rosa} (ROSA). Use the following procedures to forward the logs to AWS CloudWatch and view the logs directly through the pod by using `oc logs`.
+
+[NOTE]
+====
+ROSA is not preconfigured with a logging solution.
+====
+
+.Prerequisites
+
+* Set up a logging solution before viewing your logs. See the xref:../../observability/logging/cluster-logging-deploying.adoc#cluster-logging-deploying[Installing logging] documentation for more information.
+
+[role="_additional-resources"]
+.Additional resources
+* xref:../../observability/logging/cluster-logging.adoc#cluster-logging[Cluster logging]
+
+== Forwarding logs to CloudWatch
+
+Install the logging add-on service to forward the logs to AWS CloudWatch.
+
+//These steps were adopted from the Managed OpenShift Black Belt Tutorial "Using the AWS CloudWatch agent to publish metrics to CloudWatch in ROSA".
+
+. Run the following script to configure your ROSA cluster to forward logs to CloudWatch:
++
+[source,terminal]
+----
+$ curl https://raw.githubusercontent.com/openshift-cs/rosaworkshop/master/rosa-workshop/ostoy/resources/configure-cloudwatch.sh | bash
+----
++
+[NOTE]
+====
+Configuring ROSA to send logs to CloudWatch goes beyond the scope of this tutorial. Integrating with AWS and enabling CloudWatch logging are important aspects of ROSA, so a script is included to simplify the configuration process. The script will automatically set up AWS CloudWatch. You can examine the script to understand the steps involved.
+====
++
+.Example output
++
+[source,terminal]
+----
+Varaibles are set...ok.
+Policy already exists...ok.
+Created RosaCloudWatch-mycluster role.
+Attached role policy.
+Deploying the Red Hat OpenShift Logging Operator
+namespace/openshift-logging configured
+operatorgroup.operators.coreos.com/cluster-logging created
+subscription.operators.coreos.com/cluster-logging created
+Waiting for Red Hat OpenShift Logging Operator deployment to complete...
+Red Hat OpenShift Logging Operator deployed.
+secret/cloudwatch-credentials created
+clusterlogforwarder.logging.openshift.io/instance created
+clusterlogging.logging.openshift.io/instance created
+Complete.
+----
+. After a few minutes, you should begin to see log groups inside of AWS CloudWatch. Run the following command to see the log groups:
++
+[source,terminal]
+----
+$ aws logs describe-log-groups --log-group-name-prefix rosa-mycluster
+----
++
+.Example output
++
+[source,terminal]
+----
+{
```

Comment:
This did not work in any of my testing. I think that since this tutorial relies on Cloudwatch, we need to include a section on configuring AWS CloudWatch.

Reply:
For now, I am going to leave this as is because the prerequisite is a configured logging solution. In the future, I think this will be a really good thing to add.